src/inference/src/dev/converter_utils.cpp

```diff
@@ -397,3 +411,296 @@ std::shared_ptr<ov::ICompiledModel> ov::legacy_convert::convert_compiled_model(
+    }
+
+    void cancel() override {
+        m_request->Cancel();
+    }
+
+    std::vector<ov::ProfilingInfo> get_profiling_info() const override {
+        auto ieInfos = m_request->GetPerformanceCounts();
+        std::vector<ov::ProfilingInfo> infos;
+        infos.reserve(ieInfos.size());
+        while (!ieInfos.empty()) {
+            auto itIeInfo = std::min_element(
+                std::begin(ieInfos),
+                std::end(ieInfos),
+                [](const decltype(ieInfos)::value_type& lhs, const decltype(ieInfos)::value_type& rhs) {
+                    return lhs.second.execution_index < rhs.second.execution_index;
+                });
+            IE_ASSERT(itIeInfo != ieInfos.end());
+            auto& ieInfo = itIeInfo->second;
+            infos.push_back(ov::ProfilingInfo{});
+            auto& info = infos.back();
+            switch (ieInfo.status) {
+            case InferenceEngine::InferenceEngineProfileInfo::NOT_RUN:
+                info.status = ov::ProfilingInfo::Status::NOT_RUN;
+                break;
+            case InferenceEngine::InferenceEngineProfileInfo::OPTIMIZED_OUT:
+                info.status = ov::ProfilingInfo::Status::OPTIMIZED_OUT;
+                break;
+            case InferenceEngine::InferenceEngineProfileInfo::EXECUTED:
+                info.status = ov::ProfilingInfo::Status::EXECUTED;
+                break;
+            }
+            info.real_time = std::chrono::microseconds{ieInfo.realTime_uSec};
+            info.cpu_time = std::chrono::microseconds{ieInfo.cpu_uSec};
+            info.node_name = itIeInfo->first;
+            info.exec_type = std::string{ieInfo.exec_type};
+            info.node_type = std::string{ieInfo.layer_type};
+            ieInfos.erase(itIeInfo);
+        }
+        return infos;
+    }
+
+    ov::Tensor get_tensor(const ov::Output<const ov::Node>& port) const override {
+        const auto& name = get_legacy_name_from_port(port);
+        OPENVINO_ASSERT(!m_request->GetBlobs(name),
+                        "get_tensor shall not be used together with batched "
+                        "set_tensors/set_input_tensors for name '",
+                        name,
+                        "'");
+        auto blob = m_request->GetBlob(name);
+        // soVec = {_so, _impl->getPointerToSo()};
+        // Tensor tensor = {blob, soVec};
+        ov::Tensor tensor = {blob, {}};
+        return tensor;
+    }
+    void set_tensor(const ov::Output<const ov::Node>& port, const ov::Tensor& tensor) override {
+        m_request->SetBlob(get_legacy_name_from_port(port), tensor._impl);
+    }
+
+    std::vector<ov::Tensor> get_tensors(const ov::Output<const ov::Node>& port) const override {
+        auto blobs = m_request->GetBlobs(get_legacy_name_from_port(port));
+        std::vector<ov::Tensor> ret;
+        if (!blobs)
+            return ret;
+        for (size_t i = 0; i < blobs->size(); i++) {
+            ret.emplace_back(ov::Tensor{blobs->getBlob(i), {}});
```

Comment:
should not we also add getPointerToSo() here in some other places in this class?